lib/DefinePlugin.js

```diff
@@ -9,6 +9,10 @@ const BasicEvaluatedExpression = require("./BasicEvaluatedExpression");
 const ParserHelpers = require("./ParserHelpers");
 const NullFactory = require("./NullFactory");
 
+/** @typedef {import("./Compiler")} Compiler */
+/** @typedef {import("./Parser")} Parser */
+/** @typedef {null|undefined|RegExp|Function|Object} CodeValue */
```

Comment:
`null|undefined|RegExp|Function|string|number|Record<string, CodeValue>|RuntimeValue`

Reply:
Assuming this is not recursive I ended up with: 

```ts
/** @typedef {null|undefined|RegExp|Function|string|number} CodeValuePrimitive */
/** @typedef {CodeValuePrimitive|Record<string, CodeValuePrimitive>|RuntimeValue} CodeValue */
```